code/game/objects/items/ego_weapons/waw.dm

```diff
@@ -579,3 +579,223 @@
 
 /obj/item/ego_weapon/wings/proc/resetSpecial()
 	specialing = FALSE
+
+/obj/item/ego_weapon/mini/mirth
+	name = "mirth"
+	desc = "EGO, how does it work?"
+	special = "This weapon can be paired with its sister blade."
+	icon_state = "mirth"
+	force = 15
+	attack_speed = 0.5
+	damtype = WHITE_DAMAGE
+	armortype = WHITE_DAMAGE
+	attack_verb_continuous = list("cuts", "attacks", "slashes")
+	attack_verb_simple = list("cut", "attack", "slash")
+	hitsound = 'sound/weapons/ego/sword1.ogg'
+	attribute_requirements = list(
+							TEMPERANCE_ATTRIBUTE = 60,
+							PRUDENCE_ATTRIBUTE = 60
+							)
+
+	var/combo_on = TRUE
+	var/sound = FALSE
+
+//Switch between weapons every hit, or don't
+/obj/item/ego_weapon/mini/mirth/attack_self(mob/user)
+	..()
+	if(combo_on)
+		to_chat(user,"<span class='warning'>You swap your grip, and will no longer fight with two weapons.</span>")
+		combo_on = FALSE
+		return
+	if(!combo_on)
+		to_chat(user,"<span class='warning'>You swap your grip, and will now fight with two weapons.</span>")
+		combo_on =TRUE
+		return
+
+/obj/item/ego_weapon/mini/mirth/attack(mob/living/M, mob/living/user)
+	if(!CanUseEgo(user))
+		return
+	var/combo = FALSE
```

Comment:
Just combine the `istype` and `combo_on` check into one here.
See: `(istype(Y) && combo_on)`